### DISCUSSION TITLE
Model selection and setting when using ChatGPT.

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [acheong08/ChatGPT#976] [General]
How can I do the model selection and setting when using ChatGPT, as described [here](https://platform.openai.com/docs/api-reference/chat)?

![image](https://user-images.githubusercontent.com/11155854/222293982-80173fc2-a34b-4924-b56a-df5c99b0bc30.png)

Regards,
Zhao

**Comment by acheong08** [comment]
Use https://github.com/acheong08/ChatGPT#v3-official-chat-api

**Reply by hongyi-zhao** [reply]
Do you mean the `--temperature` option?

**Reply by acheong08** [reply]
The model is set to that by default. There are no other ChatGPT models at the moment

**Reply by hongyi-zhao** [reply]
OK. Could you please give me some hints on the meaning of some parameters of the [v3-official-chat-api](https://github.com/acheong08/ChatGPT#v3-official-chat-api):

![image](https://user-images.githubusercontent.com/11155854/222346118-86c771ec-4278-4411-a164-3f15ac3bb851.png)

**Reply by acheong08** [reply]
Temperature: Level of creativity

Stream: Whether to display the output live

Base prompt: Instructions to the model. You can ask it to act like Trump or something

**Reply by lijiazheng99** [reply]
> The model is set to that by default. There are no other ChatGPT models at the moment

Do you have any idea of which model the official API correspond to? legacy or default

**Reply by hongyi-zhao** [reply]
I think the following official documents provide more details and clues:

https://platform.openai.com/docs/models/overview
https://platform.openai.com/docs/models/gpt-3
https://platform.openai.com/docs/models/gpt-3-5
